### DISCUSSION TITLE
IPv6 default route not added on a docker instance

### DISCUSSION THREAD

**Discussion by diasdmhub** [AdguardTeam/AdGuardHome#6940] [General]
### Prerequisites

- [X] I have checked the [Wiki](https://github.com/AdguardTeam/AdGuardHome/wiki) and [Discussions](https://github.com/AdguardTeam/AdGuardHome/discussions/categories/q-a) and found no answer

- [X] I have searched other issues and found no duplicates

- [X] I want to report a bug and not [ask a question or ask for help](https://github.com/AdguardTeam/AdGuardHome/discussions/categories/q-a)

- [X] I have set up AdGuard Home correctly and [configured clients to use it](https://github.com/AdguardTeam/AdGuardHome/wiki/Clients). (Use the [Discussions](https://github.com/AdguardTeam/AdGuardHome/discussions/categories/q-a) for help with installing and configuring clients.)

### Platform (OS and CPU architecture)

Linux, ARM64

### Installation

Docker

### Setup

On a router, DHCP is handled by the router

### AdGuard Home version

0.107.48

### Action

Tested upstream servers and IPv6 was unreachable.

```sh
# ping 2800:3f0:4004:807::200e
PING 2800:3f0:4004:807::200e (2800:3f0:4004:807::200e): 56 data bytes
ping: sendto: Network unreachable
```

### Expected result

Remote IPv6 addresses should be reachable.

### Actual result

All remote IPv6 addresses are unreachable.
![ipv6_upstream_fail](https://github.com/AdguardTeam/AdGuardHome/assets/67486897/fc8ee880-ed95-4db3-b9ad-6e95d432dc71)

### Additional information and/or screenshots

I've set up an ADH container on my Mikrotik router and it works fine for IPv4 addresses, **but not for IPv6**.
As a result, **I can't reach any IPv6 upstream DNS servers.**
```
Server "https://[2001:4860:4860::8888]:443/dns-query": could not be used, please check that you've written it correctly
```

<BR>

The container has a valid IPv6 under the router.
```
# ip -6 add show dev eth0
2: eth0@if4: <BROADCAST,MULTICAST,UP,LOWER_UP,M-DOWN> mtu 1500 state UP qlen 1000
    inet6 fd08:192:168:8:9cd3:83ff:fec8:599b/64 scope global dynamic flags 100
       valid_lft 2591995sec preferred_lft 604795sec
    inet6 fd08:192:168:8::5/128 scope global
       valid_lft forever preferred_lft forever
    inet6 fe80::9cd3:83ff:fec8:599b/64 scope link
       valid_lft forever preferred_lft forever
```

<BR>

I can ping LAN IPv6 addresses, but **not remote ones**.
```
# ping 2001:4860:4860::8844
--- 2001:4860:4860::8844 ping statistics ---
5 packets transmitted, 0 packets received, 100% packet loss
```

<BR>

Then I realized that the container **does not have a default  IPv6 gateway** set up.
```
04-22 23:33:40 container,info,debug 2024/04/23 02:33:40.190436 [error] dnsproxy: upstream tls://[2620:fe::fe:10]:853 failed to exchange ;domain.name.tINt AAAA in 3.148431845s: getting conn to tls://[2620:fe::fe:10]:853: connecting to 2620:fe::fe:10: dial tcp [2620:fe::fe:10]:853: connect: no route to host
```
```
# ip -6 route show
fd08:192:168:8::5 dev eth0  metric 256
fd08:192:168:8::/64 dev eth0  metric 256  expires 0sec
fe80::/64 dev eth0  metric 256
ff00::/8 dev eth0  metric 256
```

<BR>

Only if I manually add the default route, it can reach IPv6 addresses.
```
ip -6 route add default via fd08:192:168:8::1 dev eth0 metric 1
```
**However, if the router is rebooted or the container is reset, the default route is lost**.

My assumption is that it should have set the default route dynamically.
Any idea why it is unable to add the IPv6 default route?

**Comment by ghost** [comment]
I have converted this to a discussion as this does not appear to be an issue with AdGuard Home.

I setup a test environment with a fresh install of AdGuard Home within Docker, and enabled IPv6 for a **[default bridge network](https://docs.docker.com/config/daemon/ipv6/#use-ipv6-for-the-default-bridge-network)**.
The only changes I made to my AGH instance was to set the upstream DNS to the one in your comment.

Prior to completing this, I received the same error you did. After enabling IPv6, the issue disappeared.

<img width="1904" alt="Screenshot 2024-04-26 at 12 42 26 AM" src="https://github.com/AdguardTeam/AdGuardHome/assets/149243371/6bb60dfa-f97e-4844-b456-da16d1e83688">

**Comment by diasdmhub** [comment]
The Mikrotik router has no option to configure the Docker daemon or to set up the default IPv6 bridge network.

The container interface was configured with an IPv6 gateway, but somehow it was not set as the default gateway inside the container.
```
[@MikroTik] > interface/veth/print detail
 0  R name="veth1" address=192.168.8.5/24,fd08:192:168:8::5/128 gateway=192.168.8.1 gateway6=fd08:192:168:8::1
```

**Comment by diasdmhub** [comment]
I realized that the router's container interface was configured with an incorrect IPv6 address and netmask, `fd08:192:168:8::5/128`.
After changing the netmask to `/64`, the container added the IPv6 default gateway properly.

Thank you for your attention.
